bindings/rust/src/lib.rs

```diff
@@ -151,6 +218,36 @@ impl ExecutionResult {
     }
 }
 
+/// The result of an execution.
+pub struct TypedExecutionResult {
+    result: sys::FizzyExecutionResult,
```

Comment:
Would this be more convenient as `ExecutionResult` type?

Reply:
No, because it hides the very details we are checking. Basically TypedExecutionResult and ExecutionResult is a bit of a duplication. I think in the future we may want to review this and perhaps get rid of the unsafe execution, once we have benchmarked the overheads.